File: leetcode/easy/258_add-digits/dduck-258_add-digits.py
"""
https://leetcode.com/problems/add-digits/

Runtime: 28 ms, faster than 87.12% of Python3 online submissions for Add Digits.
Memory Usage: 12.7 MB, less than 100.00% of Python3 online submissions for Add Digits.

Runtime: 24 ms, faster than 96.79% of Python3 online submissions for Add Digits.
Memory Usage: 12.9 MB, less than 100.00% of Python3 online submissions for Add Digits.
"""


class Solution:
    def addDigits(self, num: int) -> int:
        if num < 10:
            return num
        return self.addDigits(sum([int(x) for x in list(str(num))]))


print(Solution().addDigits(38))

File: leetcode/easy/258_add_digits/dduck-258_add-digits.js
// https://leetcode.com/problems/add-digits/

// Runtime: 72 ms, faster than 80.19% of JavaScript online submissions for Add Digits.
// Memory Usage: 36.1 MB, less than 66.67% of JavaScript online submissions for Add Digits.
// Runtime: 84 ms, faster than 26.57% of JavaScript online submissions for Add Digits.
// Memory Usage: 36.1 MB, less than 66.67% of JavaScript online submissions for Add Digits.

const addDigits = (num) => (num < 10 ? num
  : addDigits(num.toString().split('').reduce((acc, cur) => acc + parseInt(cur, 10), 0)));

console.log(addDigits(38));
//?
